fog/view/server/test-utils/src/lib.rs

```diff
@@ -369,3 +443,71 @@ pub fn wait_for_server_to_load(
         sleep(Duration::from_secs(1));
     }
 }
+
+/// Wait until a server has added a specific number of blocks to load.
+pub fn wait_for_block_to_load(block_count: u64, store_servers: &[TestViewServer], logger: &Logger) {
+    let mut allowed_tries = 60usize;
+    loop {
+        let server_num_blocks = get_highest_processed_block_count(store_servers);
+        if server_num_blocks >= block_count {
+            break;
+        }
+        log::info!(
+            logger,
+            "Waiting for server to catch up to db... {} < {}",
+            server_num_blocks,
+            block_count,
+        );
+        if allowed_tries == 0 {
+            panic!("Server did not catch up to database!");
+        }
+        allowed_tries -= 1;
+        sleep(Duration::from_millis(1000));
+    }
+}
+
+/// Wait until a server has added a specific number of blocks to load.
+pub fn wait_for_highest_processed_and_last_known(
+    view_client: &mut FogViewGrpcClient,
+    highest_processed_block_count: u64,
+    last_known_block_count: u64,
+) {
+    let mut allowed_tries = 60usize;
+    loop {
+        let nonsense_search_keys = vec![vec![50u8]];
+        let result = view_client.request(0, 0, nonsense_search_keys).unwrap();
+        if result.highest_processed_block_count == highest_processed_block_count
+            && result.last_known_block_count == last_known_block_count
+        {
+            break;
+        }
+
+        if allowed_tries == 0 {
+            panic!("Server did not catch up to database! highest_processed_block_count = {}, last_known_block_count = {}", result.highest_processed_block_count, result.last_known_block_count);
+        }
+        allowed_tries -= 1;
+        sleep(Duration::from_millis(1000));
+    }
+}
+
+/// Find the highest processed number of blocks in a collection of store
+/// servers.
+pub fn get_highest_processed_block_count(store_servers: &[TestViewServer]) -> u64 {
+    store_servers
+        .iter()
+        .map(|server| server.highest_processed_block_count())
+        .max()
+        .unwrap_or_default()
+}
+
+/// Creates a list of BlockRanges for store servers.
+pub fn create_block_ranges(store_count: usize, blocks_per_store: u64) -> Vec<BlockRange> {
+    let mut store_block_ranges = Vec::new();
+    for i in 0..store_count {
+        let start_block = (i as u64) * blocks_per_store;
+        let block_range = BlockRange::new_from_length(start_block, blocks_per_store);
+        store_block_ranges.push(block_range);
+    }
+
+    store_block_ranges
```

Comment:
:upside_down_face: I'm not sure it's any better, at first glance I was hoping there was some kind of chunking or similar that might make it nicer to iterate, but I'm unsure if`(a * b)` and the `step_by()` isn't more obfuscated
```suggestion
    let step = blocks_per_store as usize;
    (0..store_count * step)
        .step_by(step)
        .map(|i| BlockRange::new_from_length(i as u64, blocks_per_store))
        .collect::<Vec<_>>()
```

Reply:
Done.